packages/flutter_tools/lib/src/test/test_wrapper.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:test_api/backend.dart'; // ignore: deprecated_member_use
+import 'package:test_core/src/runner/platform.dart'; // ignore: implementation_imports
+import 'package:test_core/src/executable.dart' as test; // ignore: implementation_imports
+import 'package:test_core/src/runner/hack_register_platform.dart' as hack; // ignore: implementation_imports
+
+export 'package:test_api/backend.dart' show Runtime; // ignore: deprecated_member_use
+export 'package:test_core/src/runner/platform.dart' show PlatformPlugin; // ignore: implementation_imports
+
+abstract class TestWrapper {
+  const TestWrapper();
```

Comment:
The way we do this in a few other places is:

```
abstract class TestWrapper {
  factory TestWrapper() => _DefaultTestWrapper();
  ...
}

class _DefaultTestWrapper implements TestWrapper {
  ...
}
```

There are a couple of reasons for this:
1. It adds an additional layer that prevents the rest of the tool relying on APIs of the implementation class that are not in the interface.
2. It prevents code that does src/ imports of the tool code extending/implementing the implementation class, which turns many kinds of useful changes into breaking changes.

Reply:
This doesn't allow for const construction of a `TestWrapper` though, and doesn't allow it to be defaulted in an argument list that initializes a final variable.  Will play with that a little.

Reply:
Going with a static const field - I think that achieves the same goal while still making subclasses possible and const construction possible.

Reply:
A redirecting factory constructor can be const (at least according to the spec):
```dart
abstract class TestWrapper {
  const factory TestWrapper() = _DefaultTestWrapper
}
class _DefaultTestWrapper implements TestWrapper {
  const _DefaultTestWrapper();
}
```

Reply:
Ah! Doing the weird factory assignment thing works better.